### ISSUE TITLE
assertion failed

### ISSUE THREAD

**Issue by kehiy** [essentialkaos/aligo#89]
we are trying to use aligo in our project linter, but we face with this issue:

```
aligo --tags "" check ./...
panic: /opt/hostedtoolcache/go/1.20.7/x64/src/go/types/sizes.go:181: assertion failed

goroutine 1 [running]:
go/types.assert(0x18?)
```

the [faild workflow](https://github.com/pactus-project/pactus/actions/runs/5833961924/job/15822490032?pr=632)

how can we fix it?

**Comment by andyone** [comment]
You get this error because OOM killer killed the process of _aligo_ due to lack of memory. Your project has a huge code base and building AST for analysis exceeds the size of GitHub actions worker memory. You can try setting memory limits like that:

```bash
GOMEMLIMIT=5MiB aligo check ./...
```
But anyway, checking for such a big amount of code on small worker (building AST requires parsing all dependencies at any depth) will take 30-40 minutes or even more.

**Comment by andyone** [comment]
@kehiy JFYI I've released _aligo_ **[2.0.0](https://kaos.sh/aligo/2.0.0)**. I've replaced the deprecated `go/loader` package with `go/packages` and now it uses less memory and inspects sources much faster. It should help using _aligo_ with your project without memory limits.

**Comment by kehiy** [comment]
@andyone 
thank you so much I will test it!
(but the last time I faced with the same issue with 25MIB mem limit)
